addons/suggest_locations/models/stock_move_line.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+from collections import defaultdict
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+from .suggest_locations_policy import SUGGEST_LOCATION_REGISTRY
+
+DROPABLE_STATES = ("assigned", "partially_available")
+CONTRAINTS_REQUIRING_A_CHECK = ("enforce", "enforce_with_empty")
+WITH_EMPTY_LOCATIONS = ("suggest_with_empty", "enforce_with_empty")
+VIEW_SET = set(["view"])
+
+
+class MoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _get_policy_class(self, picking_type, preprocessing=False):
+        """Get the policy for suggesting locations
+        Note: Optionally we could cache the results with @lru_cache to avoid always returning the
+        policy class, but this caused issued with closed cursors. To avoid this,
+        you need to update the environment at the same time you return the cached result.
+        Possibly by having another function that returns the cached class and updated environment
+        by policy.env = self.env
+        """
+        try:
+            policy_name = picking_type.u_suggest_locations_policy
+            policy_class = SUGGEST_LOCATION_REGISTRY[policy_name]
+            return policy_class(self.env, preprocessing)
+        except KeyError:
+            raise ValueError(
+                _(f"Policy with name={picking_type.u_suggest_locations_policy} could not be found")
+            )
+
+    def suggest_locations(self, picking=None, values=None, limit=30):
+        """Suggest locations for move line, either by self or picking_type and values"""
+        # Validate preconditions
+        if not self and (not picking or not values):
+            raise ValueError(
+                _(
+                    "Missing information! If suggest locations is not done via self, picking "
+                    + "and values must be provided!"
+                )
+            )
+        # Set the picking and the policy_type
+        if self and not picking:
+            picking = self.picking_id
+            policy_type = self.u_picking_type_id
```

Comment:
I think there has been a misunderstanding withe my previous comments since this naming is confusing, this should be `picking_type`, and then get the policy like `policy_type = policy_type.u_suggest_locations_policy` and use policy_type straightaway in the if:  `if (not policy_type or policy_type == "None"): `